### DISCUSSION TITLE
How to compress PSD file from 65MB to 15MB .ZIP file without loss of quality?

### DISCUSSION THREAD

**Discussion by graphics777** [aonez/Keka#1350] [Q&A] [answered]
I would like to learn how to compress a .psd file (Photoshopfile with smartlayers) sized aprox 65-70MB down to 15MB .zipfile with Keka?

I know it is possile, I just don’t know how it was done. And I am hoping somebody here on this forum can help me how it is possible.

Background: I have received a .zip file (sized 15MB) that contained a compressed Photoshop file (.PSD file with 4 smartlayers inside), that was sized 65MB when I decompressed the .zip file.

A compression of -75%: From 65MB psd file to 15MB .zip file.

So I know this level of compression is possible, I just don't know how it was done, and I need to learn to do the same.

Now, how is this possible to do this level of compression in keka with the .ZIP format? 

Are there some ‘secret’ settings in Keka, that I don’t know about? So far I have only been able to compress the same file down to 36MB, a -44% compression.

I need to get the .zip file down under 20MB as this is the upload limit I am working with.

So In short: How do I compress a 65MB .PSD file down to a 15MB .zip file, without loss of quality of the file.

NB I know I can get much lower compression with fx the Z7 file format (i managed getting it down to 18MB with Z7 compression), but it has to be a .ZIP format, because it has to be an easy fileformat for everyone to open without special software.

**Comment by gingerbeardman** [comment, marked as answer]
if you still have the original fil you can see how it was compressed. on command line do: `7z l -slt file.zip`

that said, compression is not predictable like this. What I mean by that is your 65MB may not compress as well as my 65MB - it depends on the content.

but in keka you control zip size using the slider (slow = smallest)

<img width="448" alt="Screen shot 2023-10-20 at 13 49 10" src="https://github.com/aonez/Keka/assets/49612/0d189979-445a-431a-8351-3f95b23bb879">

**Comment by graphics777** [comment]
Hello Matt

Thank you for your answer.
I have tried compressing the same (original) 65MB .PSD file (that I
received as a compressed .ZIP file sized 15MB) with Keka on the 'Slowest'
setting, as a .ZIP file, but I can only get it down to 36MB this way in the
zip format.
It is just a mystery to me how it was originally compressed down to 15MB in
the .ZIP format (it is .ZIP NOT 7ZIP).
I do not have the possibility of asking the person who originally
compressed the file so I have to learn it by myself, and I am trying my
best to solve it by asking experts who know more about this than me.

I completely understand your point that it depends on the content, and not
all files will compress to the same extend, but this is the same file, so I
am not doing it optimally, and I woul really like to find out how it was
done.

Could you explain more in detail, or refer me to some info/guide on how I
can see how the original file was compressed? I unfortunately do not
understand how to do what you suggest I do, (on command line do: 7z l -slt
file.zip)

I need a little bit more explanation :-)

PS I know that I can compress it way more with the 7Zip format, using Keka
(got it down to 18MB) but it need to be compressed as a .ZIP format, so
that all my customers can open the file without special software.

Thank you!

**Reply by gingerbeardman** [reply]
can I download the original PSD anywhere? Then I can answer the question for you!

To see zip info, this assumes the PSD file in in your home directory and not in any sub-folders.

0. If you don't have a separate 7z you'll need it
1. Open Terminal.app
2. Type `7z l -slt original.zip`
3. The output of the command will show you how each file in the zip is compressed

For each file in the zip you'll get output like this, which tells us the compressions settings used:

```
Path = crogine-golf-v1.12.1/windows/resource.h
Folder = -
Size = 387
Packed Size = 227
Modified = 2023-04-23 14:31:11
Created = 
Accessed = 
Attributes = 
Encrypted = -
Comment = 
CRC = E2F5C2C2
Method = Deflate
Characteristics = UT:M:1
Host OS = FAT
Version = 10
Volume Index = 0
Offset = 306264399
```

**Reply by graphics777** [reply]
Hi Matt,

Thank you very much for the detailed info on how to see how the original file was compressed. Unfortunately I do not have 7Z installed and cant follow your instructions.

I hope that it is understood from my help request that it is a .ZIP file (**not** a 7zip file) that I am trying to find out how was compressed -75%.

It is really important for me to find out how to compress large PSD file to this extend (from 65 MB PSD file to 15 MB .zip file)
The whole mystery to me is how this PSD file was compressed from 65MB to 15MB .Zip file.
So far I have only been able to get this level of compression with the 7zip format, but I need to compress my files with .ZIP not 7Zip, so that the people I share the file with, will be able to open the file without special software.

So it needs to be compressed as .ZIP not 7zip.

Would you be able to tell me how the original psd file was compressed if you get acces to the original .Zip file of 15MB?
What information will you be able to see? Can you see software was used to compress it? Or what info about the compression will be available?

Thank you.

**Comment by graphics777** [comment]
Hi Matt, 

Thank you very much for the detailed info on how to see how the original file was compressed. Unfortunately I do not have 7Z installed and cant follow your instructions.

It is really important for me to find out how to compress large PSD file to this extend (from 65 MB PSD file to 15 MB .zip file)
The whole mystery to me is how this PSD file was compressed from 65MB to 15MB .Zip file.
So far I have only been able to get this level of compression with the 7zip format, but I need to compress my files with .ZIP not 7Zip, so that the people I share the file with, will be able to open the file without special software. 

So it needs to be compressed as .ZIP not 7zip.

Would you be able to tell me how the original psd file was compressed if you get acces to the original .Zip file of 15MB?
What information will you be able to see? Can you see software was used to compress it? Or what info about the compression will be available?

Thank you.

**Reply by aonez** [reply]
> Unfortunately I do not have 7Z installed and cant follow your instructions.

You can use Keka on the Terminal.app for that:

```sh
/Applications/Keka.app/Contents/MacOS/Keka --cli 7zz  l -slt original.zip
```
